src/lib/Folder.interface.ts

```diff
@@ -0,0 +1,401 @@
+  createdAt: Date;
+  id: string;
+  internalComments: InternalComment[];
+  metadata: { extension: string; mimeType: string; name: string; size: number };
+  properties: DocumentProperties;
+  rejectionCode: number;
+  signedUrl: string;
+  signedUrlCreatedAt: Date;
+  signedUrlTTL: number;
+  state: FileState;
+  updatedAt: Date;
+}
+
+/**
+ * Legal file
+ */
+export interface LegalFile extends File {
+  type: LegalFileType;
+}
+
+/**
+ * Supporting file
+ */
+export interface SupportingFile extends File {
+  type: SupportingFileType;
+}
+
+/**
+ * State of the file
+ */
+export enum FileState {
+  IN_PROGRESS = 1,
+  REJECTED,
+  SUSPECTED,
+  OK,
+}
+
+/**
+ * Type of the file
+ */
+export enum FileType {
+  CONTRACT = 'CONTRACT',
+  INFORMATION = 'INFORMATION',
+  CONVENTION_OF_PROOF = 'CONVENTION_OF_PROOF',
+  LOAN_AND_PAYMENT_AGREEMENT = 'LOAN_AND_PAYMENT_AGREEMENT',
+  TERMS_OF_SALE = 'TERMS_OF_SALE',
+  SEPA = 'SEPA',
+  BANK_RECORDS = 'BANK_RECORDS',
+  LEGAL_MENTIONS = 'LEGAL_MENTIONS',
+  RECEIPT = 'RECEIPT',
+  OTHERS = 'OTHERS',
+}
+
+/**
+ * Document period
+ */
+export interface DocumentPeriod {
+  count: number;
+  unit: DocumentUnitPeriod;
+}
+
+/**
+ * Unit period of document
+ */
+export enum DocumentUnitPeriod {
+  YEAR = 'YEAR',
+  MONTH = 'MONTH',
+  WEEK = 'WEEK',
+  DAY = 'DAY',
+}
+
+/**
+ * Legal file type
+ */
+export enum LegalFileType {
+  CONTRACT = 'CONTRACT',
+  INFORMATION = 'INFORMATION',
+  CONVENTION_OF_PROOF = 'CONVENTION_OF_PROOF',
+  LOAN_AND_PAYMENT_AGREEMENT = 'LOAN_AND_PAYMENT_AGREEMENT',
+  TERMS_OF_SALE = 'TERMS_OF_SALE',
+  SEPA = 'SEPA',
+  BANK_RECORDS = 'BANK_RECORDS',
+  LEGAL_MENTIONS = 'LEGAL_MENTIONS',
+  RECEIPT = 'RECEIPT',
+  OTHERS = 'OTHERS',
+}
+
+/**
+ * State of the document
+ */
+export enum DocumentState {
+  CREATED,
+  IN_PROGRESS,
+  REJECTED,
+  SUSPECTED,
+  OK,
+}
+
+/**
+ * Category of legal document
+ */
+export enum LegalDocumentCategory {
+  INSURANCE_CONTRACT = 'INSURANCE_CONTRACT',
+  PRODUCT_CONTRACT = 'PRODUCT_CONTRACT',
+  DEBIT_MANDATE = 'DEBIT_MANDATE',
+  RECORDS = 'RECORDS',
+  LEGAL_MENTIONS = 'LEGAL_MENTIONS',
+  INFORMATION = 'INFORMATION',
+  RECEIPT = 'RECEIPT',
+  OTHERS = 'OTHERS',
+}
+
+/**
+ * Signature
+ */
+export interface ISignature {
+  callbackUrl: string;
+  createdAt: Date;
+  holder: Holder;
+  id: string;
+  legalDocumentIds: string[];
+  metadata: unknown;
+  partnerId: string;
+  redirectUrl: string;
+  redirectUrlCreatedAt: Date;
+  redirectUrlTTL: number;
+  state: SignatureState;
+}
+
+/**
+ * State of the signature
+ */
+export enum SignatureState {
+  CREATED,
+  IN_PROGRESS,
+  REJECTED,
+  // eslint-disable-next-line no-magic-numbers
+  SIGNED = 4,
+}
+
+/**
+ * Rejection code
+ */
+/* eslint-disable no-magic-numbers */
+export enum RejectionCode {
+  OUTSIDE_PERIMETER = 100,
+  FORGED = 101,
+  UNREADABLE = 102,
+  NOT_COHERENT = 103,
+  EXPIRED = 104,
+  INCOMPLETE = 105,
+}
+/* eslint-disable no-magic-numbers */
```

Comment:
I was forced to disable `no-magic-numbers` rule here. But I think that in this context, the rule is not relevant.

Reply:
I agree, we should adapt our [eslint rule](https://eslint.org/docs/rules/no-magic-numbers)